Add new /sys/well-known interface to get information about registered labels (#25695)
hashicorp/vault: website/content/api-docs/system/well-known.mdx

---
layout: api
page_title: /sys/well-known - HTTP API
description: The `/sys/well-known` endpoint is used to list and read registered .well-known labels in Vault.
---

# `/sys/well-known`

> The `/sys/well-known` endpoint is used to list and read the registered labels within the server's .well-known/ path prefix

Reviewer: ```suggestion
Use the `/sys/well-known` endpoint to list and read registered labels on the `.well-known/` path prefix for your Vault instance.
```
Style correction: write in active voice.

## List well-known

> This endpoint lists all registered labels.

Reviewer: ```suggestion
List all registered labels.
```
Style correction: avoid "this" as a pronoun

| Method | Path              |
|:-------|:------------------|
| `GET`  | `/sys/well-known` |
| `LIST` | `/sys/well-known` |

### Sample request

```shell-session
$ curl \
    --header "X-Vault-Token: ..." \
    http://127.0.0.1:8200/v1/sys/well-known
```

### Sample response

```json
{
  "key_info": {
    "est/cacerts": {
      "mount_path": "ns1/pki_int/",
      "mount_uuid": "fc9d3ee4-ae92-4e3e-c0e1-a1fdb3e3b8cf",
      "prefix": "est/cacerts"
    }
  },
  "keys": [
    "est/cacerts"
  ]
}
```

## Read well-known

> This endpoint retrieves the information around a specific registered label

Reviewer: ```suggestion
Retrieve information for the specified label.
```
Style correction: avoid "this" as a pronoun

| Method | Path                     |
| :----- |:-------------------------|
| `GET`  | `/sys/well-known/:label` |

### Parameters

> - `label` `(string: <required>)` – Specifies the registered label to lookup
>   This is specified as part of the request URL.

Reviewer: ```suggestion
- `label` `(string: <required>)` – URL path parameter specifying the registered label to fetch.
```
Style correction: avoid "this" as a pronoun
